contrib/gcppubsub/pkg/controller/channel/reconcile.go

```diff
@@ -169,100 +144,70 @@ func (r *reconciler) reconcile(ctx context.Context, c *eventingv1alpha1.Channel)
-		return false, err
+		recorder.Eventf(c, v1.EventTypeWarning, gcpResourcesPlanFailed, "Failed to plan Channel's resources: %v", err)
+		return false, reconcile.Result{}, err
 	}
 	if persist == persistStatus {
 		if err = pubsubutil.SetInternalStatus(ctx, c, plannedPCS); err != nil {
-			r.recorder.Eventf(c, v1.EventTypeWarning, gcpResourcesPersistFailed, "Failed to persist Channel's resources: %v", err)
-			return false, err
+			recorder.Eventf(c, v1.EventTypeWarning, gcpResourcesPersistFailed, "Failed to persist Channel's resources: %v", err)
+			return false, reconcile.Result{}, err
 		}
 		// Persist this and run another reconcile loop to enact it.
-		return true, nil
+		return false, reconcile.Result{}, err
 	}
 
 	svc, err := r.createK8sService(ctx, c)
 	if err != nil {
-		r.recorder.Eventf(c, v1.EventTypeWarning, k8sServiceCreateFailed, "Failed to reconcile Channel's K8s Service: %v", err)
-		return false, err
+		recorder.Eventf(c, v1.EventTypeWarning, k8sServiceCreateFailed, "Failed to reconcile Channel's K8s Service: %v", err)
+		return false, reconcile.Result{}, err
 	}
 
 	err = r.createVirtualService(ctx, c, svc)
 	if err != nil {
-		r.recorder.Eventf(c, v1.EventTypeWarning, virtualServiceCreateFailed, "Failed to reconcile Virtual Service for the Channel: %v", err)
-		return false, err
+		recorder.Eventf(c, v1.EventTypeWarning, virtualServiceCreateFailed, "Failed to reconcile Virtual Service for the Channel: %v", err)
+		return false, reconcile.Result{}, err
 	}
 
 	topic, err := r.createTopic(ctx, plannedPCS, gcpCreds)
 	if err != nil {
-		r.recorder.Eventf(c, v1.EventTypeWarning, topicCreateFailed, "Failed to reconcile Topic for the Channel: %v", err)
-		return false, err
+		recorder.Eventf(c, v1.EventTypeWarning, topicCreateFailed, "Failed to reconcile Topic for the Channel: %v", err)
+		return false, reconcile.Result{}, err
 	}
 
 	err = r.syncSubscriptions(ctx, plannedPCS, gcpCreds, topic, subsToSync)
 	if err != nil {
-		r.recorder.Eventf(c, v1.EventTypeWarning, subscriptionSyncFailed, "Failed to reconcile Subscription for the Channel: %v", err)
-		return false, err
+		recorder.Eventf(c, v1.EventTypeWarning, subscriptionSyncFailed, "Failed to reconcile Subscription for the Channel: %v", err)
+		return false, reconcile.Result{}, err
 	}
 	// Now that the subs have synced successfully, remove the old ones from the status.
 	plannedPCS.Subscriptions = subsToSync.subsToCreate
 	if err = pubsubutil.SetInternalStatus(ctx, c, plannedPCS); err != nil {
-		r.recorder.Eventf(c, v1.EventTypeWarning, subscriptionDeleteFailed, "Failed to delete old Subscriptions from the Channel's status: %v", err)
-		return false, err
+		recorder.Eventf(c, v1.EventTypeWarning, subscriptionDeleteFailed, "Failed to delete old Subscriptions from the Channel's status: %v", err)
+		return false, reconcile.Result{}, err
 	}
 
 	c.Status.MarkProvisioned()
-	return false, nil
+	return false, reconcile.Result{}, nil
```

Comment:
The boolean is wrong. The old code has it meaning 'requeue'. Now it means update the status. It should be `true` here (and probably almost everywhere else too).

Reply:
Thats is correct. Have already fixed it